packages/markdown/src/index.ts

```diff
@@ -0,0 +1,315 @@
+import type { FileType, Service, FileChangeType } from '@volar/language-service';
+import MarkdownIt from 'markdown-it';
+import { Emitter } from 'vscode-jsonrpc';
+import type { TextDocument } from 'vscode-languageserver-textdocument';
+import type { ILogger, IMdLanguageService, IMdParser, IWorkspace } from 'vscode-markdown-languageservice';
+import { createLanguageService, DiagnosticLevel, githubSlugifier, LogLevel } from 'vscode-markdown-languageservice';
+import { URI } from 'vscode-uri';
+
+export interface Provide {
+	'markdown/languageService': () => IMdLanguageService;
+}
+
+const md = new MarkdownIt();
+
+function isMarkdown(document: TextDocument): boolean {
+	return document.languageId === 'markdown';
+}
+
+function assert(condition: unknown, message: string): asserts condition {
+	if (!condition) {
+		throw new Error(message);
+	}
+}
+
+export function create(): Service<Provide> {
+	return (context) => {
+		if (!context) {
+			return {} as any;
+		}
+
+		let lastProjectVersion = context.host.getProjectVersion();
+		assert(context.env, 'context.env must be defined');
+		const { fs, onDidChangeWatchedFiles } = context.env;
+		assert(fs, 'context.env.fs must be defined');
+		assert(
+			onDidChangeWatchedFiles,
+			'context.env.fs.onDidChangeWatchedFiles must be defined'
+		);
+
+		const logger: ILogger = {
+			level: LogLevel.Off,
+
+			log(_logLevel, message) {
+				context.env.console?.log(message);
+			}
+		};
+
+		const parser: IMdParser = {
+			slugifier: githubSlugifier,
+
+			async tokenize(document) {
+				return md.parse(document.getText(), {});
+			}
+		};
+
+		const onDidChangeMarkdownDocument = new Emitter<TextDocument>();
+		const onDidCreateMarkdownDocument = new Emitter<TextDocument>();
+		const onDidDeleteMarkdownDocument = new Emitter<URI>();
+
+		const fileWatcher = onDidChangeWatchedFiles((event) => {
+			for (const change of event.changes) {
+				switch (change.type) {
+					case 2 satisfies typeof FileChangeType.Changed: {
+						const document = context.getTextDocument(change.uri);
+						if (document) {
+							onDidChangeMarkdownDocument.fire(document);
+						}
+						break;
+					}
+
+					case 1 satisfies typeof FileChangeType.Created: {
+						const document = context.getTextDocument(change.uri);
+						if (document) {
+							onDidCreateMarkdownDocument.fire(document);
+						}
+						break;
+					}
+
+					case 3 satisfies typeof FileChangeType.Deleted: {
+						onDidDeleteMarkdownDocument.fire(URI.parse(change.uri));
+						break;
+					}
+				}
+			}
+		});
+
+		const workspace: IWorkspace = {
+			async getAllMarkdownDocuments() {
+				return [];
```

Comment:
Is this an intentional non-implementation? It seems to just to execute `sync()` and return `syncedVersions.values()`.

Reply:
You’re right. Initially I skipped this before `sync()` was implemented.